snowflake/objects/deploy.sql

```diff
@@ -0,0 +1,77 @@
+/*
+  This is the primary deployment script for RECOVER Snowflake objects.
+
+  Deployments are made to a specific environment. An environment is analogous
+  to a database. Account-level objects are not covered by this script. We assume
+  that the following objects have already been created at the account level:
+
+    - ROLE `RECOVER_DATA_ENGINEER`
+        * A ROLE with the appropriate privelages to create and deploy
+          objects in a database.
+    - WAREHOUSE `RECOVER_XSMALL`
+    - API INTEGRATION `RECOVER_GIT`
+        * An API INTEGRATION with an API_ALLOWED_PREFIXES value
+          containing 'https://github.com/Sage-Bionetworks/' and
+          API_PROVIDER = GIT_HTTPS_API.
+    - STORAGE INTEGRATION `RECOVER_PROD_S3`
+        * An S3 storage integration which allows access to the
+          S3 buckets in the RECOVER production account.
+
+  Additionally, we assume that the following databases have already been created
+  when deploying to the "staging" or "main" environment, respectively:
+
+    - DATABASE `RECOVER_STAGING`
+    - DATABASE `RECOVER_MAIN`
+
+  To deploy to an environment identified by your current git branch,
+  we can invoke this file with the Snowflake CLI like so:
+
+  `snow sql -D "environment=$(git rev-parse --abbrev-ref HEAD)" -f {this file}`
+*/
+
+/*
+  Configure the runner
+*/
+USE ROLE RECOVER_DATA_ENGINEER;
+USE WAREHOUSE RECOVER_XSMALL;
+
+/*
+  Configure and bootstrap the database if this is a dev environment.
+
+  Since we typically want to use the branch name as the environment identifier,
+  but only A-Za-z0-9 and _ and $ characters are allowed in Snowflake identifier
+  names and we have the convention of naming the branch after its associated
+  Jira ticket (e.g., etl-123), we substitute a likely hyphen in the `environment`
+  variable passed at runtime.
+*/
+SET safe_environment_identifier = (SELECT REPLACE('&{ environment }', '-', '_'));
+SET database_identifier = CONCAT('recover_', $safe_environment_identifier);
+
+EXECUTE IMMEDIATE
+$$
+BEGIN
+    IF ('&{ environment }' != 'main' AND '&{ environment }' != 'staging') THEN
```

Comment:
I'm probably missing this somewhere but since we don't have a `staging` branch and I don't see us having an env variable in GH workflow for `staging`, how does it know we are running staging?

Reply:
Hmm this is an oversight on my part.

Currently, merging into `main` triggers a deployment to our `staging` namespace in AWS. But as it's written now, merging into `main` will trigger a deployment to our `main` environment in Snowflake.

I can think of two solutions off the top of my head:
1. Change the process to match the Snowflake deployment code. This entails doing all PRs into a `staging` branch and then deploying this branch to both `staging` AWS and Snowflake environments.
2. Change the Snowflake deployment code to match the process. Change the logic in our top-level `deploy.sql` so that merging into main does a deployment to the `staging` Snowflake environment as well. Update the `upload-and-deploy-to-prod-main` workflow to do a deployment to the `main` Snowflake environment.

(2) seems significantly simpler to me, so I'd prefer to implement that solution.

Reply:
I agree with (2), and it matches our current deployment workflow for AWS stacks.